dot/rpc/modules/author_test.go

```diff
@@ -14,173 +14,145 @@ import (
-	}
-
-	var res HasSessionKeyResponse
-	err := module.HasSessionKeys(nil, req, &res)
-	require.NoError(t, err)
-	require.False(t, bool(res))
-
-	coremockapi.AssertCalled(t, "DecodeSessionKeys", mock.AnythingOfType("[]uint8"))
-}
-
-func TestAuthorModule_HasSessionKey_WhenRuntimeFails(t *testing.T) {
-	coremockapi := new(apimocks.CoreAPI)
-	coremockapi.On("DecodeSessionKeys", mock.AnythingOfType("[]uint8")).Return(nil, errors.New("problems with runtime"))
+func TestAuthorModule_HasSessionKeys(t *testing.T) {
+	testReq := "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d34309a9d2a24213896ff06895db16" +
+		"aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc38520426026343" +
+		"09a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc38520426026"
+	pkeys := common.MustHexToBytes("0x0102d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d34309a9d2a24213896ff06895db16" +
+		"aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aa" +
+		"de8b6502f3a71cf56374cc38520426026")
+	data := common.MustHexToBytes("0x011080d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d6772616e8034309a9d2a24213896f" +
+		"f06895db16aade8b6502f3a71cf56374cc38520426026626162658034309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc38520426026696d6f6e8034" +
+		"309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602661756469")
 
-	module := &AuthorModule{
-		coreAPI: coremockapi,
-		logger:  log.New(log.SetWriter(io.Discard)),
-	}
+	coreMockAPIDecodeErr := new(apimocks.CoreAPI)
+	coreMockAPIDecodeErr.On("DecodeSessionKeys", []byte{0x4, 0x1}).Return(nil, errors.New("decodeSessionKeys err"))
 
-	req := &HasSessionKeyRequest{
-		PublicKeys: "0x00",
-	}
+	coreMockAPIUnmarshalErr := new(apimocks.CoreAPI)
+	coreMockAPIUnmarshalErr.On("DecodeSessionKeys", []byte{0x4, 0x1}).Return([]byte{0x4, 0x1}, nil)
 
-	var res HasSessionKeyResponse
-	err := module.HasSessionKeys(nil, req, &res)
-	require.Error(t, err, "problems with runtime")
-	require.False(t, bool(res))
-}
+	coreMockAPIOk := new(apimocks.CoreAPI)
+	coreMockAPIOk.On("DecodeSessionKeys", pkeys).Return(data, nil)
+	coreMockAPIOk.On("HasKey", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(true, nil)
 
-func TestAuthorModule_HasSessionKey_WhenThereIsNoKeys(t *testing.T) {
-	keys := "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d34309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc38520426026"
-	runtimeInstance := wasmer.NewTestInstance(t, runtime.NODE_RUNTIME)
+	coreMockAPIErr := new(apimocks.CoreAPI)
+	coreMockAPIErr.On("DecodeSessionKeys", pkeys).Return(data, nil)
+	coreMockAPIErr.On("HasKey", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(false, errors.New("HasKey err"))
```

Comment:
Let's not have huge strings

```suggestion
	coreMockAPIErr.
		On("HasKey", mock.AnythingOfType("string"), mock.AnythingOfType("string")).
		Return(false, errors.New("HasKey err"))
```